cl/lib/courts.py

```diff
@@ -0,0 +1,135 @@
+from dataclasses import dataclass
+
+from django.core.cache import cache
+
+from cl.search.models import Court
+
+
+def get_cache_key_for_court_list() -> str:
+    """
+    Retrieves the cache key used to store the list of courts.
+
+    This function provides a consistent and easily identifiable key for caching
+    court data. It's particularly useful for:
+
+    -   Ensuring consistent cache access across different parts of the
+        application.
+    -   Facilitating cache isolation during testing, allowing for predictable
+        and controlled test environments.
+    -   Simplifying cache invalidation and management.
+
+    :return: The cache key, which is currently "minimal-court-list".
+    """
+    return "minimal-court-list"
+
+
+@dataclass
+class MinimalCourtData:
+    pk: str
+    short_name: str
+    in_use: bool
+    parent_court_id: str | None
+
+
+def get_minimal_list_of_courts() -> list[MinimalCourtData]:
+    """
+    Retrieves a list of courts with minimal data.
+
+    This method fetches a list of courts from the database, including only the
+    primary key (pk), short name, in_use status, jurisdiction, and parent court
+    ID. It uses a cache to store and retrieve this data, reducing db load.
+
+    The cache is set to expire after 24 hours.
+
+    :return: A list of MinimalCourtData objects
+    """
+
+    data = cache.get(get_cache_key_for_court_list())
+    if data is not None:
+        return data
+
+    courts = list(
+        Court.objects.values("pk", "short_name", "in_use", "parent_court_id")
+    )
+    court_list = [MinimalCourtData(**court) for court in courts]
+    cache.set(get_cache_key_for_court_list(), court_list, 60 * 60 * 24)
+
+    return court_list
+
+
+def get_active_court_from_cache() -> list[MinimalCourtData]:
+    """
+    Retrieves a list of active courts from the cached court data.
+
+    This method fetches the court data using `get_minimal_list_of_courts()`. it
+    then filters this list to include only courts where the 'in_use' attribute
+    is True.
+
+    :returns: A list of MinimalCourtData objects representing active courts.
+    """
+    courts = get_minimal_list_of_courts()
+    return [c for c in courts if c.in_use]
+
+
+def get_parent_ids_from_cache(courts: list[MinimalCourtData]) -> set[str]:
```

Comment:
This doesn't actually involve the cache, so maybe:

```suggestion
def get_court_parent_ids(courts: list[MinimalCourtData]) -> set[str]:
```

Reply:
yeah, makes sense. 👍